core/CMakeLists.txt

```diff
@@ -154,6 +154,21 @@ if(THEROCK_ENABLE_HIP_RUNTIME)
       libhiprtc-builtins.so
       libhiprtc.so
   )
+  therock_cmake_subproject_declare(hip-tests
+    USE_DIST_AMDGPU_TARGETS
+    EXTERNAL_SOURCE_DIR "${THEROCK_ROCM_SYSTEMS_SOURCE_DIR}/projects/hip-tests/catch"
```

Comment:
I also wonder if we'd want to run other samples from hip-tests (https://github.com/ROCm/rocm-systems/tree/develop/projects/hip-tests/samples), or if the `catch/` folder has enough representative tests. The samples did look more standalone and harder to integrate into this build system...